src/wallet/Wallet.ts

```diff
@@ -134,6 +152,16 @@ class Wallet {
     this.balance = this.utxoSet.totalBalance - this.pending.amount;
   }
 
+  /**
+   * Updates the selected network
+   * @param network name of the network
+   */
+  async updateNetwork(network: SelectedNetwork): Promise<void> {
```

Comment:
As a part of this function, we should clear out the rest of the state of the wallet:
the utxo set should be cleared `this.utxoSet.clear()`
the address set should be cleared (`this.addressManager = new AddressManager()`)
pending transactions should be cleared (`this.pending.transactions = {}`)
and transactions should be cleared (`this.transactions = []; this.transactionsStorage = {}`)

And a test should be written making sure that everything has been cleared and that things work as normal after the fact. I can def help with this!

Reply:
I'm fine with either of these approaches, I'll defer to you and which one you prefer

Reply:
We discussed this yesterday, but I'm fine with/in favor of this approach. I think it's ok if changing your network is expensive from a performance standpoint.

> As an alternative we can blow out the entire wallet, create a new one from same seed, and add stuff to the constructor/static methods to make sure we initialize with the new network setting.

Reply:
State reset added in 0382e6c53fefa6e34a422ed00d4f9a152a39de55 (I didn't add tests though)